### ISSUE TITLE
Sheduled news do not trigger RSS Updates

### ISSUE THREAD

**Issue by kroerig** [contao/contao#6572]
### Affected version(s)

4.13

### Description

Steps to reproduce:
1. Create a RSS Feed for a news category.
2. Create a new news entry and shedule for some time in the future.
3. Wait for the Cron-Job to aktivate it
4. Have a look into the RSS feed

It takes up to 24h for the entry to appear in the RSS Feed. That's too long.

**Comment by fritzmg** [comment]
The same probably happens in Contao 5.

**Comment by Toflar** [comment]
I don't think we can fix this without re-thinking the start/stop logic. We have several bug reports related to this.

**Comment by kroerig** [comment]
But maybe you can change the cron job that updates the xml files to run hourly instead of daily.

**Comment by Toflar** [comment]
There is no XML file. It's a response that's cached in the reverse proxy so it's not a simple as that. Yes, we need to work with cronjobs somehow but we need to completely re-think the start/stop logic because **all** these elements have that issue.

**Comment by kroerig** [comment]
On https://docs.contao.org/dev/framework/cron/ it says there is a cron job
![grafik](https://github.com/contao/contao/assets/5148795/d8eba3ba-bbdf-4fb0-86a2-3c32a7d5ba5c)

**Comment by Toflar** [comment]
It works completely different in Contao 5 where this problem would persist. Anyway, I just wanted to say is that I don't see any way to fix that anytime soon. That's all.

The docs are partially outdated, I'll fix that.

**Comment by fritzmg** [comment]
> On https://docs.contao.org/dev/framework/cron/ it says there is a cron job

Yes, in Contao 4.13. If you want the XML to be updated daily, you can do so with your own Cronjob if you want.

**Comment by aschempp** [comment]
we can fix this in Contao 5 imho, but we cannot fix it in Contao 4.13 without majorly changing stuff (which I wouldn't do for an LTS version shortly before expiration). So I suggest to keep this issue open for Contao 5 until someone finds some time to fix this 😊

**Comment by leofeyer** [comment]
> we can fix this in Contao 5 imho,

What exactly do you want to fix? In Contao 5, news feeds are page controllers, so they should automatically show timed news once they become public, shouldn‘t they?

**Comment by fritzmg** [comment]
No, because that would require active cache invalidation - which does not happen.

**Comment by leofeyer** [comment]
Of course you have to select a matching cache timeout in the page settings. If you publish daily, you cannot cache for longer than 24 hours. If you publish hourly, you cannot cache for longer than 60 minutes and so on. I don‘t see what could be "fixed" about this.

**Comment by aschempp** [comment]
We could (and should) set the page cache automatically to the next expiring and/or upcoming news. Same as we already clear the page cache if a news is changed.

**Comment by kroerig** [comment]
anything new on this with the upcoming 5.3?

**Comment by leofeyer** [comment]
No, unfortunately not. We discussed this again in today‘s call and came to the conclusion that an automation service could perhaps solve this problem (and would also open up many new possibilities for us). But implementing something like this is not trivial.

**Comment by fritzmg** [comment]
We have developed a sort of proof-of-concept extension that utilizes the `DelayStamp` for Symfony Messenger messages in order to invalidate the cache at a certain point in the future: https://github.com/inspiredminds/contao-future-cache-invalidation

So when any DCA with a `start` or `stop` field is saved, such a message is generated and dispatched over the message bus. However, this solution is still imperfect, as there are no cache tags one could invalidate for `start` - since the element was not rendered and thus also no cache tag exists for it. Instead the extension tries to invalidate the parent record in that case. And if no parent record is available it simply clears the whole cache (like if you have set a `start` date for a specific page, then the whole cache will need to be cleared at that time anyway in order to let it be shown on in the navigation module on all pages).

_Note:_ the usage of the `DelayStamp` is only supported by the following messenger transport types:

* `doctrine`
* `amqp`
* `redis`

**Comment by aschempp** [comment]
shouldn't this be rather simple for news feeds like this? https://github.com/contao/contao/pull/8301
